Explorer/Assets/DCL/Notifications/NotificationsMenu/NotificationsMenuController.cs

```diff
@@ -113,8 +123,29 @@ private void OnViewShown()
             }
         }
 
-        private async UniTaskVoid InitialNotificationRequestAsync(CancellationToken ct)
+        private async UniTaskVoid CheckIdentityChangeAsync(CancellationToken token)
         {
+            if (previousWeb3Identity != null)
+                await InitialNotificationRequestAsync(token);
+
+            while (token.IsCancellationRequested == false && panelWasOpenedOnce == false)
```

Comment:
Instead of using a while loop here, maybe it's more clear to use `UniTask.WaitUntil`:
```
await UniTask.WaitUntil(() => token.IsCancellationRequested || panelWasOpenedOnce);

if (token.IsCancellationRequested)
    return;
```

Reply:
maybe I'm missing something but here the purpose of the while loop is to cycle every `IDENTITY_CHANGE_POLLING_INTERVAL` (line 139) in order to check the identity change and trigger the logic accordingly.
I am not passively waiting for `panelWasOpenedOnce` (or the cancellation token) in order to proceed, which is how I translate the suggestion

Reply:
Oh sorry, I saw the `while` and I directly though it was a "waiting" loop 👍 Ok, forget my comment!